contracts/eosiolib/varint.hpp

```diff
@@ -6,42 +6,207 @@
 
 
 /**
- * @defgroup varint Variable Length Integer
+ * @defgroup varint Variable Length Integer Type
+ * @brief Defines variable length integer type which provides more efficient serialization
  * @ingroup types
- * @{
+ * @{/
  */
 
+/**
+ * Variable Length Unsigned Integer. This provides more efficient serialization of 32-bit unsigned int.
+ * It serialuzes a 32-bit unsigned integer in as few bytes as possible
+ * 
```

Comment:
This is probably an appropriate place to define the encoding.

refer to https://github.com/EOSIO/eos/issues/1950

not sure why that issue was closed, maybe it was moved to another repo